packages/wallet/src/KeyManagement/LedgerKeyAgent.ts

```diff
@@ -0,0 +1,128 @@
+import { Cardano } from '@cardano-sdk/core';
+import { DeviceCommunicationType, createDeviceConnection, establishDeviceConnection } from './util/deviceConnection';
+import { GenericError, GenericErrorType, TransportError } from './errors';
+import { GroupedAddress, KeyAgentType, SerializableLedgerKeyAgentData, SignBlobResult } from './types';
+import { KeyAgentBase } from './KeyAgentBase';
+import AppAda, { GetVersionResponse, utils } from '@cardano-foundation/ledgerjs-hw-app-cardano';
+
+export interface LedgerKeyAgentProps {
+  networkId: Cardano.NetworkId;
+  accountIndex: number;
+  knownAddresses: GroupedAddress[];
+  deviceConnection: AppAda;
+  deviceCommunicationType: DeviceCommunicationType;
+}
+
+export enum TransportType {
+  WebHid = 'webhid',
+  NodeHid = 'nodehid'
```

Comment:
```suggestion
```

Reply:
I added this and decided to leave it because we have a guard against unsupported type. But, I can remove it if you think that is a better idea

Reply:
I say drop this until it's implemented, since this is a model of our supported transport types.